backend/faq/models.py

```diff
@@ -28,6 +30,55 @@ class Meta:
     def __str__(self) -> str:
         return self.name
 
+    def get_text(self, related="locales", lang=None):
+        """
+        Return the translated text of a related instance.
+
+        A related instance can be a question or an answer.
+        """
+        if lang is None:
+            lang = translation.get_language()
+        try:
+            locale = getattr(self, related).get(language=lang)
+        except ObjectDoesNotExist:
+            try:
+                locale = getattr(self, related).get(language="en")
+            except ObjectDoesNotExist:
+                return self.name  # pylint: disable=no-member
+        return locale.text
+
+    def get_text_prefetch(self, related="locales", lang=None):
+        """
+        Return the translated text of a related instance.
+
+        Contrary to `self.get_text` this method consider the related instances
+        as already prefetched with `prefetch_related`, and use `.all()` instead
+        of `.get()` to avoid triggering any additional SQL query.
+        """
+        if lang is None:
+            lang = translation.get_language()
+
+        try:
+            locale = [loc for loc in getattr(self, related).all()
+                      if loc.language == lang][0]
+
+        except IndexError:
+            try:
+                locale = [loc for loc in getattr(self, related).all()
+                          if loc.language == "en"][0]
+
+            except IndexError:
+                return self.name  # pylint: disable=no-member
+        return locale.text
+
+    def get_question_text_prefetch(self, lang=None):
+        """Return the translated text of the question."""
+        return self.get_text_prefetch(related="locales", lang=lang)
```

Comment:
Reading this asymmetry where questions are called "locales" and answers are called "answers" makes me think that we may rename the main model "FAQEntry" which contains locales for questions and answers (very minor)

Reply:
Good idea.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1130/commits/f90ddf65b20a5648e84582c27f012f2df5b3c3bb